3p/iframe-messaging-client.js

```diff
@@ -116,7 +125,20 @@ export class IframeMessagingClient {
   }
 
   /**
-   * @param {!Window} win
+   * Determines whether frameOrWindow is a frame, or a window.
+   * @returns {boolean}
+   */
+  hostWindowIsActuallyAnIframe_() {
+    // If it's a window, it will have .postMessage
+    // We check for that before .contentWindow, since a cross-domain window
+    // may throw if we try to access anything unsafe
+    return (!this.hostWindow_.postMessage && !!this.hostWindow_.contentWindow);
```

Comment:
I don't believe we actually need to test for postMessage as any AMP supported browser would have it.